src/libraries/System.Text.Json/src/System/Text/Json/ThrowHelper.Serialization.cs

```diff
@@ -26,6 +26,13 @@ public static void ThrowNotSupportedException_SerializationNotSupported(Type pro
             throw new NotSupportedException(SR.Format(SR.SerializationNotSupportedType, propertyType));
         }
 
+        [DoesNotReturn]
+        [MethodImpl(MethodImplOptions.NoInlining)]
+        public static void ThrowNotSupportedException_CannotDeserializeOpenGeneric(Type type)
```

Comment:
Is `[DoesNotReturn]` on void method redundant?

Reply:
No. DoesNotReturn isn't about return values. It tells nullability flow analysis that the method literally won't return, e.g. it'll crash or throw or exit the process or some such thing, but code immediately after the call site is unreachable for the purposes of analyses.